src/main/java/org/craftercms/studio/impl/v1/service/security/SecurityServiceImpl.java

```diff
@@ -187,22 +187,25 @@ public Set<String> getUserPermissions(@ValidateStringParam(name = "site") final
             addGroupRoles(roles, site, groups, rolesConfig);
             // resolve the permission
             permissions = populateUserPermissions(site, path, roles, permissionsConfig);
-            // check if the user is allowed to edit the content
+            logger.trace("Check if the user is allowed to edit the content in site '{}' path '{}' user '{}' " +
+                            "groups '{}'", site, path, user, groups);
 
+            // TODO: SJ: refactor the code below if it's still in use, otherwise remove
             if (path.indexOf("/site") == 0) { // If it's content a file
                 try {
                     ContentTypeConfigTO config = contentTypeService.getContentTypeForContent(site, path);
                     boolean isAllowed = contentTypeService.isUserAllowed(roles, config);
                     if (!isAllowed) {
-                        logger.debug("The user is not allowed to access " + site + ":" + path
-                                + ". adding permission: " + StudioConstants.PERMISSION_VALUE_NOT_ALLOWED);
+                        logger.trace("User '{}' is not permitted to access site '{}' path '{}', adding " +
+                                "permission '{}'", user, site, path, StudioConstants.PERMISSION_VALUE_NOT_ALLOWED);
                         // If no default role is set
                         permissions.add(StudioConstants.PERMISSION_VALUE_NOT_ALLOWED);
                         return permissions;
                     }
                 } catch (ServiceLayerException e) {
-                    logger.debug("Error while getting the content type of " + path
-                            + ". skipping user role checking on the content.");
+                    // TODO: SJ: Is this really a debug?
```

Comment:
Discuss with AV and JM.

Reply:
Warning at least IMO. No reason why a content item would not have a content type.
For content not found scenario, it should resolve and return 404 before reaching this method.